packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -300,8 +304,36 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
 
   private void handleChoosePictureResult(int resultCode, Intent data) {
     if (resultCode == Activity.RESULT_OK && data != null) {
-      String path = fileUtils.getPathFromUri(activity, data.getData());
-      handleResult(path);
+        //String path = fileUtils.getPathFromUri(activity, data.getData());
+
+        //creating a temp file from uri to allow reading pictures from remote sources
+        try {
+            InputStream inputStream = null;
+            OutputStream output = null;
+            try {
+                inputStream = activity.getContentResolver().openInputStream(data.getData());
+                File file = new File(activity.getCacheDir(), "cachedImage.jpg");
```

Comment:
What if that file already exists?
When is the file cleaned up?

Reply:
I guessed that for the cleaning since it is the cacheDir and I was trying to always use the same file name that was ok. In that sense imageResizer is even worse since it create a scaled image version in the external directory where the original file is, leaving the user with two copies of it. However you are right about the file already existing